x/profiles/keeper/msg_server_app_link.go

```diff
@@ -19,6 +22,18 @@ func (k Keeper) LinkApplication(
 		return nil, err
 	}
 
+	// See if the link already exists
+	link, found, err := k.GetApplicationLink(ctx, msg.Sender, msg.LinkData.Application, msg.LinkData.Username)
+	if err != nil {
+		return nil, err
+	}
+
+	if found && link.IsVerificationOngoing() {
```

Comment:
Shouldn't we also check if the link has already been verified?

Reply:
The ability to start the verification all-over has been left open for two scenarios: 

1. The user has verified the link, and the link validity has later expired (in the future, app links will have a limited validity)
2. The verification process has ended with an error. 

In both cases, the link would appear as the verification has ended, but the user must be able to start from scratch with the same link.

Reply:
@RiccardoM Cool! Thanks for the clarification.